packages/base/Button/src/ButtonSplit/styles.ts

```diff
@@ -1,59 +1,51 @@
-import type { Theme } from '@material-ui/core/styles'
-import { createStyles } from '@material-ui/core/styles'
-
-export default ({ palette }: Theme) =>
-  createStyles({
-    actionButton: {
-      borderTopRightRadius: 0,
-      borderBottomRightRadius: 0,
-    },
-    menuButton: {
-      minWidth: '2em',
-      paddingLeft: '0.5em',
-      paddingRight: '0.5em',
-      borderTopLeftRadius: 0,
-      borderBottomLeftRadius: 0,
-      marginLeft: 0,
-    },
-
-    primaryVariant: {
-      '&$actionButton': {
-        borderRight: `1px solid ${palette.blue.darker}`,
-      },
-      '&$menuButton': {
-        borderLeft: `1px solid ${palette.blue.darker}`,
-
-        '&$disabled': {
-          borderLeftColor: palette.grey.main,
-        },
-      },
-    },
-
-    smallSize: {
-      minWidth: '1.5em',
-      paddingLeft: '0.25em',
-      paddingRight: '0.25em',
-    },
-    mediumSize: {},
-    largeSize: {
-      minWidth: '3em',
-      paddingLeft: '0.75em',
-      paddingRight: '0.75em',
-    },
-
-    dropdown: {
-      display: 'block',
-      cursor: 'pointer',
-    },
-
-    disabled: {
-      // override dropdown anchor styles
-      '& > div': {
-        cursor: 'auto',
-      },
-    },
-
-    rotated: {
-      transform: 'rotate(180deg)',
-    },
-  })
+import type { SizeType } from '@toptal/picasso-shared'
+
+export const createActionButtonClassNames = ({
+  variant,
+}: {
+  variant?: 'primary' | 'secondary'
+}) => {
+  const classNames: string[] = ['transition-[color,background]']
+
+  if (variant === 'primary') {
+    classNames.push(
+      'border-r border-l-0 border-y-0 border-solid border-blue-700'
+    )
+  }
+
+  return classNames
+}
+
+const menuButtonSizeClassNames: Record<
+  SizeType<'small' | 'medium' | 'large'>,
+  string
+> = {
+  small: 'min-w-[1.5em] px-[0.25em]',
+  medium: 'min-w-[2em] px-[0.5em]',
+  large: 'min-w-[3em] px-[0.75em]',
+}
+
+export const createMenuButtonClassNames = ({
+  variant,
+  size,
+  disabled,
+}: {
+  variant?: 'primary' | 'secondary'
+  size: SizeType<'small' | 'medium' | 'large'>
+  disabled?: boolean
+}) => {
+  const classNames: string[] = []
+
+  if (variant === 'primary') {
+    classNames.push('border-l border-r-0 border-y-0 border-solid')
+    if (disabled) {
+      classNames.push('border-gray-500')
+    } else {
+      classNames.push('border-blue-700')
+    }
+  }
+
+  classNames.push(menuButtonSizeClassNames[size])
+
+  return classNames
+}
```

Comment:
Maybe we could simplify these with [classnames](https://www.npmjs.com/package/classnames), instead of doing  the handling by hand, wdyt? classnames is kinda of a industry standard for these types of problems, and the preferred way of doing it on Totpal
```suggestion
}) => cx('transition-[color,background]', {
  'border-r border-l-0 border-y-0 border-solid border-blue-700': variant === 'primary'
})

const menuButtonSizeClassNames: Record<
  SizeType<'small' | 'medium' | 'large'>,
  string
> = {
  small: 'min-w-[1.5em] px-[0.25em]',
  medium: 'min-w-[2em] px-[0.5em]',
  large: 'min-w-[3em] px-[0.75em]',
}

export const createMenuButtonClassNames = ({
  variant,
  size,
  disabled,
}: {
  variant?: 'primary' | 'secondary'
  size: SizeType<'small' | 'medium' | 'large'>
  disabled?: boolean
}) => cx({
  'border-l border-r-0 border-y-0 border-solid': variant === 'primary',
  'border-gray-500': variant === 'primary' && disabled,
  'border-gray-700': variant === 'primary' && disabled,
}, menuButtonSizeClassNames[size])
```

Reply:
Just for awareness: https://github.com/dcastil/tailwind-merge/discussions/137#discussioncomment-3481605